be/src/olap/push_handler.cpp

```diff
@@ -372,7 +391,11 @@ Status PushBrokerReader::next(vectorized::Block* block) {
     if (!_ready || block == nullptr) {
         return Status::Error<INVALID_ARGUMENT>();
     }
-    _scanner->get_next(block, &_eof);
+    if (_cur_reader == nullptr || _cur_reader_eof) {
+        RETURN_IF_ERROR(_get_next_reader());
+    }
```

Comment:
if _eof=true, get_next_reader will return ok. but _cur_reader == null, it will core at line 398.